### PR TITLE
feat(parser): Add Markdown support and comprehensive tests for markdo…

### PR DESCRIPTION
**Pull request by Waazify**
…wn tests parsing

- Introduced support for parsing Markdown elements including headers, lists, blockquotes, code blocks, and tables in the parser.
- Enhanced the TestCaseElement interface to accommodate Markdown metadata.
- Updated the TestCaseParser to handle Markdown formatting and preserve structure during parsing.
- Added a new example file for code review with suggestions for code improvements.
- Implemented extensive unit tests to validate Markdown parsing functionality, covering various scenarios and edge cases.

These changes improve the parser's capabilities and ensure robust handling of Markdown content in test cases.

This PR close issue #34 

## Summary by CodeRabbit

- **New Features**
  - Added support for parsing and preserving a wide range of Markdown formatting within test case messages, including headers, lists, code blocks, blockquotes, tables, and inline styles.
  - Test files with both `.txt` and `.md` extensions are now supported.

- **Bug Fixes**
  - Improved handling of test case titles with both single and double hash Markdown headers.
  - Enhanced validation logic for system messages in test cases.

- **Tests**
  - Introduced comprehensive tests to verify Markdown parsing and correct handling of complex Markdown constructs in messages.

- **Documentation**
  - Added an example conversation demonstrating a code review with best practices and unit tests.

### CODE REVIEW

File: src/parser/tokenizer.ts [hunk 1]
```diff
@@ -7,12 +7,34 @@ export interface TestCaseElement {
...
 
   /**
    * The value of the element
    */
   value: string;
+
+  /**
+   * Optional metadata for the element
+   */
+  metadata?: {
```
Comment (turisanapo) [inline src/parser/tokenizer.ts:20]: What is using this metadata?

File: src/parser/markdown-handlers.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+ * Interface for markdown format handlers
+ */
+export interface MarkdownHandler {
+  pattern: RegExp;
+  handle(match: RegExpMatchArray): string;
+}
+
+/**
+ * Handler for markdown headers (h1-h6)
+ */
```
[the remark this answers is not in the document]
Comment (turisanapo) [reply src/parser/markdown-handlers.ts:26]: Checking the code, I can only see it used to produce the metadata. Is it used somewhere else?

File: src/parser/tokenizer.ts [hunk 3]
```diff
@@ -128,18 +183,62 @@ export class TestCaseParser {
...
+              value: currentContent.join('\n'),
+            });
+            currentContent = [];
+          }
           handle(line, elements);
           handled = true;
-          // Update current role if this was a role marker
           if (pattern === TestCaseParser.PATTERNS.ROLE) {
-            const match = line.match(/^\s*(\w+):/i);
+            const match = line.match(/^[\s]*(\w+):/i);
```
[the remark this answers is not in the document]
Comment (turisanapo) [reply src/parser/tokenizer.ts]: This is valid, can you address it please?

File: src/parser/tokenizer.ts (hunk 1 above)
Comment (turisanapo) [reply src/parser/tokenizer.ts]: This?

File: src/parser/tokenizer.ts [hunk 4]
```diff
@@ -7,12 +7,34 @@ export interface TestCaseElement {
   /**
    * The type of the element
    */
-  type: 'role' | 'content' | 'tool_use' | 'tool_response' | 'args';
+  type: 'role' | 'content' | 'tool_use' | 'tool_response' | 'args' | 'markdown';
```
[the remark this answers is not in the document]
Comment (turisanapo) [reply src/parser/tokenizer.ts]: We don't want this

**Review by turisanapo — APPROVED** [review]
Lgtm
